xclim/subset.py

```diff
@@ -164,6 +165,56 @@ def func_checker(*args, **kwargs):
     return func_checker
 
 
+def check_latlon_dimnames(func):
+    @wraps(func)
+    def func_checker(*args, **kwargs):
+        """
+        A decorator examining the names of the latitude and longitude dimensions and renames them temporarily if needed.
+        """
+
+        if range(len(args)) == 0:
+            return func(*args, **kwargs)
+
+        formatted_args = list()
+        conv = dict()
+        for argument in args:
+            try:
+                if isinstance(argument, (xarray.DataArray, xarray.Dataset)):
+                    dims = argument.dims
+                else:
+                    raise TypeError
+            except TypeError:
+                logging.error(f"No file or no dimensions found in arg `{argument}`.")
+                formatted_args.append(argument)
+                continue
+
+            if not {"lon", "lat"}.issubset(dims) or not {"rlon", "rlat"}.issubset(dims):
+                if {"long"}.issubset(dims):
+                    conv["long"] = "lon"
+                elif {"latitude", "longitude"}.issubset(dims):
+                    conv["latitude"] = "lat"
+                    conv["longitude"] = "lon"
+                elif {"lats", "lons"}.issubset(dims):
+                    conv["lats"] = "lat"
+                    conv["lons"] = "lon"
+                if not conv:
+                    warnings.warn(
+                        f"lat and lon-like dimensions are not found among arg `{argument}` dimensions: {list(dims)}."
+                    )
+                argument = argument.rename(conv)
+
+            formatted_args.append(argument)
+
+        final = func(*formatted_args, **kwargs)
```

Comment:
Are there cases where the output is more than one dataset? Do we want to cover them ?

Reply:
Right now I'm thinking of the FWI where latitude is needed and the core indicator function returns 6 datasets. But this is out of scope, I agree.